components/camel-olingo4/camel-olingo4-api/src/test/java/org/apache/camel/component/olingo4/Olingo4AppAPITest.java

```diff
@@ -405,6 +414,70 @@ public void testBatchRequest() throws Exception {
         assertEquals(HttpStatusCode.NOT_FOUND.getStatusCode(), responseParts.get(7).getStatusCode());
     }
 
+    @Test
+    public void testUnboundActionRequest() throws Exception {
+        final TestOlingo4ResponseHandler<HttpStatusCode> responseHandler = new TestOlingo4ResponseHandler<>();
+        olingoApp.action(edm, TEST_UNBOUND_ACTION_RESETDATASOURCE, null, null, responseHandler);
+
+        final HttpStatusCode statusCode = responseHandler.await();
+        assertEquals(204, statusCode.getStatusCode());
+    }
+
+    @Test
+    public void testBoundActionRequest() throws Exception {
+        final ClientEntity clientEntity = objFactory.newEntity(null);
+        clientEntity.getProperties().add(objFactory.newPrimitiveProperty("userName", objFactory.newPrimitiveValueBuilder().buildString("scottketchum")));
+        clientEntity.getProperties().add(objFactory.newPrimitiveProperty("tripId", objFactory.newPrimitiveValueBuilder().buildInt32(0)));
+
+        final TestOlingo4ResponseHandler<HttpStatusCode> responseHandler = new TestOlingo4ResponseHandler<>();
+        olingoApp.action(edm, TEST_BOUND_ACTION_PEOPLE_SHARETRIP, null, clientEntity, responseHandler);
+
+        final HttpStatusCode statusCode = responseHandler.await();
+        assertEquals(204, statusCode.getStatusCode());
+    }
+
+
+    // Unfortunately there is no action that returns a client entity. So we fake one
+    @Test
+    public void testBoundActionRequestWithClientEntityResponse() throws Exception {
+        final ODataClient odataClient = ODataClientFactory.getClient();
+        final ODataWriter odataWriter = odataClient.getWriter();
+
+        final HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        httpClientBuilder.addInterceptorFirst(new HttpResponseInterceptor() {
+            @Override
+            public void process(HttpResponse response, HttpContext context) throws HttpException, IOException {
+                if (response.getStatusLine().getStatusCode() == 204) {
+                    try {
+                        response.setEntity(
+                                new InputStreamEntity(
+                                        odataWriter.writeEntity(createEntity(), ContentType.JSON),
+                                        org.apache.http.entity.ContentType.parse(ContentType.JSON.toContentTypeString())));
+                        response.setStatusCode(200);
```

Comment:
It would be nice to change values to the HttpStatusCode... constants for better understanding

Reply:
Done